src/c_wrapper.jl

```diff
@@ -29,14 +29,32 @@ function SCS_solve(T::Union{Type{Direct}, Type{Indirect}},
         m::Int, n::Int, A::SCSVecOrMatOrSparse, b::Array{Float64},
         c::Array{Float64}, f::Int, l::Int, q::Array{Int}, s::Array{Int},
         ep::Int, ed::Int, p::Array{Float64},
-        primal_sol::Vector{Float64}=Float64[],
-        dual_sol::Vector{Float64}=Float64[],
-        slack::Vector{Float64}=Float64[];
+        primal_sol::Vector{Float64}=zeros(n),
+        dual_sol::Vector{Float64}=zeros(m),
+        slack::Vector{Float64}=zeros(m);
         options...)
 
     n > 0 || throw(ArgumentError("The number of variables in SCSModel must be greater than 0"))
     m > 0 || throw(ArgumentError("The number of constraints in SCSModel must be greater than 0"))
 
+    ws = (:warm_start=>true) in options
+    warmstart_sizes_are_correct = length(primal_sol) == n && length(dual_sol) == length(slack) == m
+
+    if warmstart_sizes_are_correct
+        if !ws
+            primal_sol = fill!(primal_sol, 0.0)
+            dual_sol = fill!(dual_sol, 0.0)
+            slack = fill!(slack, 0.0)
+        end
+    else
+        ws && throw(ArgumentError("The provided warmstart doesn't match problem sizes; discarding!"))
```

Comment:
"discarding" is inaccurate. The execution will stop here.

Reply:
thanks for catching this, fixed